bp-dprk-and-elections/js/scatter.js

```diff
@@ -0,0 +1,310 @@
+let yearData = [];
+
+let yearArray = [];
+let allYears = [];
+
+let dataPoints = [];
+let dataMarkers = [];
+
+let electionYearAndDays = {};
+let mins = [];
+let maxes = [];
+
+Highcharts.data({
+  // Load Data in from Google Sheets
+  googleSpreadsheetKey: "11nBD55d0t4QE1h-OeymUg897azDIC6CHME2UenQ8WSw",
+  googleSpreadsheetWorksheet: 2,
+  switchRowsAndColumns: true,
+  parsed: function parsed(columns) {
+    columns.shift();
+    columns.forEach((row) => {
+      const electionDate = row[0];
+      let electionYear = electionDate.slice(0, 4);
+      let electionYearNumber = parseFloat(electionYear);
+      const electionType = row[1];
+      const provocationDescription = row[2];
+      const numberOfDays = row[3];
+      let numberOfDaysNumber = parseFloat(numberOfDays);
+      const provocationDate = row[5];
+
+      if (electionYear >= "1990") {
+        let color = "";
+
+        if (electionType == "General / Presidential") {
+          color = "#FFC726";
+        } else {
+          color = "#10355F";
+        }
+
+        dataPoints.push({
+          x: numberOfDaysNumber,
+          y: electionYearNumber,
+          electionType: electionType,
+          provocationDescription: provocationDescription,
+          provocationDate: provocationDate,
+          electionDate: electionDate,
+          color: color,
+        });
+
+        if (!yearArray.includes(electionYearNumber)) {
+          yearArray.push(electionYearNumber);
+
+          dataMarkers.push({
+            x: 0,
+            y: electionYearNumber,
+            electionDate: electionDate,
+            color: "#F55536",
+            marker: { symbol: "triangle" },
+          });
+        }
+        if (!electionYearAndDays[electionYearNumber]) {
+          electionYearAndDays[electionYearNumber] = [];
+        }
+        electionYearAndDays[electionYearNumber].push(numberOfDays);
+      }
+    });
+
+    yearArray.forEach((year) => {
+      let min = Math.min(...electionYearAndDays[year]);
+      if (!mins.includes(min)) {
+        mins.push(min);
+      }
+    });
+
+    yearArray.forEach((year) => {
+      let max = Math.max(...electionYearAndDays[year]);
+      if (!maxes.includes(max)) {
+        maxes.push(max);
+      }
+    });
+
+    for (i = 0; i < yearArray.length; i++) {
+      if (mins[i] && maxes[i] < 0) {
+        yearData.push({
+          x: mins[i],
+          x2: 0,
+          y: yearArray[i],
+          year: yearArray[i],
+          color: "lightGray",
+        });
+      } else if (mins[i] > 0) {
+        yearData.push({
+          x: 0,
+          x2: maxes[i],
+          y: yearArray[i],
+          year: yearArray[i],
+          color: "lightGray",
+        });
+      } else {
+        yearData.push({
+          x: mins[i],
+          x2: maxes[i],
+          y: yearArray[i],
+          year: yearArray[i],
+          color: "lightGray",
+        });
+      }
+    }

```

Comment:
We can refactor this a little bit by doing something like the following.

```
for (i = 0; i < yearArray.length; i++) {
        y: yearArray[i],
        year: yearArray[i],
        color: "lightGray",
        if (mins[i] && maxes[i] < 0) {
           yearData.push({
           x: mins[i],
           x2: 0,
        });
        } else if (mins[i] > 0) {
           yearData.push({
           x: 0,
           x2: maxes[i],
        });
        } else {
           yearData.push({
           x: mins[i],
           x2: maxes[i],
        });
    }
  }
```

Reply:
On the one hand, I see what you did here. On the other, when I tried it, it broke the chart and I'm not sure why.

Reply:
Whoops! I missed a pretty major part of it 

```suggestion
    for (i = 0; i < yearArray.length; i++) {
      yearData.push({
        y: yearArray[i],
        year: yearArray[i],
        color: "lightGray",
      });
      if (mins[i] && maxes[i] < 0) {
        yearData.push({
          x: mins[i],
          x2: 0,
        });
      } else if (mins[i] > 0) {
        yearData.push({
          x: 0,
          x2: maxes[i],
        });
      } else {
        yearData.push({
          x: mins[i],
          x2: maxes[i],
        });
      }
    }
```